src/components/sbb-datepicker-next-day/sbb-datepicker-next-day.tsx

```diff
@@ -24,88 +12,97 @@ import {
 
   /** Whether the component is disabled due date equals to max date. */
-  @State() private _disabled = false;
+  @state() private _disabled = false;
 
   /** Whether the component is disabled due date-picker's input disabled. */
-  @State() private _inputDisabled = false;
+  @state() private _inputDisabled = false;
 
   /** The maximum date as set in the date-picker's input. */
-  @State() private _max: string | number;
+  @state() private _max: string | number;
 
-  @State() private _currentLanguage = documentLanguage();
+  @state() private _currentLanguage = documentLanguage();
 
   private _handlerRepository = new HandlerRepository(
-    this._element as HTMLElement,
+    this as HTMLElement,
     actionElementHandlerAspect,
     languageChangeHandlerAspect((l) => {
       this._currentLanguage = l;
       this._setAriaLabel();
     }),
   );
 
-  private _datePickerElement: HTMLSbbDatepickerElement;
+  private _datePickerElement: SbbDatepicker;
 
   private _dateAdapter: DateAdapter<Date> = new NativeDateAdapter();
 
   private _datePickerController: AbortController;
 
-  @Watch('datePicker')
-  public async findDatePicker(
-    newValue: string | HTMLElement,
-    oldValue: string | HTMLElement,
-  ): Promise<void> {
+  private _findDatePicker(newValue: string | HTMLElement, oldValue: string | HTMLElement): void {
     if (newValue !== oldValue) {
-      await this._init(this.datePicker);
+      this._init(this.datePicker);
     }
   }
+  private _abort = new ConnectedAbortController(this);
 
-  @Listen('click')
-  public async handleClick(): Promise<void> {
-    if (!this._datePickerElement || isValidAttribute(this._element, 'data-disabled')) {
+  private _handleClick(): void {
+    if (!this._datePickerElement || isValidAttribute(this, 'data-disabled')) {
       return;
     }
-    const startingDate: Date = (await this._datePickerElement.getValueAsDate()) ?? this._now();
+    const startingDate: Date = this._datePickerElement.getValueAsDate() ?? this._now();
     const date: Date = findNextAvailableDate(
       startingDate,
       this._datePickerElement.dateFilter,
       this._dateAdapter,
       this._max,
     );
     if (this._dateAdapter.compareDate(date, startingDate) !== 0) {
-      await this._datePickerElement.setValueAsDate(date);
+      this._datePickerElement.setValueAsDate(date);
     }
   }
 
-  public async connectedCallback(): Promise<void> {
+  public override connectedCallback(): void {
+    super.connectedCallback();
+    const signal = this._abort.signal;
+    this.addEventListener('click', () => this._handleClick(), { signal });
     this._handlerRepository.connect();
     this._syncUpstreamProperties();
-    await this._init(this.datePicker);
+    this._init(this.datePicker);
+  }
+
+  public override willUpdate(changedProperties: PropertyValues<this>): void {
+    if (changedProperties.has('datePicker')) {
+      this._findDatePicker(this.datePicker, changedProperties.get('datePicker'));
```

Comment:
```suggestion
      this._init(this.datePicker);
```
I don't think that the `_findDatePicker` method is needed anymore since it only checks if the value has changed (and you are already sure it is changed here)

Another possible problem is that you are calling the `_init` method twice on initialization, once in the connectedCallback and once here. Are you ok with that?

Reply:
I tried with a breakpoint, it doesn't reach the `_init` inside the willUpdate on initialization